api/v1/atlasintegration_types.go

```diff
@@ -0,0 +1,187 @@
+// +kubebuilder:validation:XValidation:rule="!has(self.microsoftTeams) || (self.type == 'MICROSOFT_TEAMS' && has(self.microsoftTeams))",message="only MICROSOFT_TEAMS type may set microsoftTeams fields"
+// +kubebuilder:validation:XValidation:rule="!has(self.newRelic) || (self.type == 'NEW_RELIC' && has(self.newRelic))",message="only NEW_RELIC type may set newRelic fields"
+// +kubebuilder:validation:XValidation:rule="!has(self.opsGenie) || (self.type == 'OPS_GENIE' && has(self.opsGenie))",message="only OPS_GENIE type may set opsGenie fields"
+// +kubebuilder:validation:XValidation:rule="!has(self.prometheus) || (self.type == 'PROMETHEUS' && has(self.prometheus))",message="only PROMETHEUS type may set prometheus fields"
+// +kubebuilder:validation:XValidation:rule="!has(self.pagerDuty) || (self.type == 'PAGER_DUTY' && has(self.pagerDuty))",message="only PAGER_DUTY type may set pagerDuty fields"
+// +kubebuilder:validation:XValidation:rule="!has(self.slack) || (self.type == 'SLACK' && has(self.slack))",message="only SLACK type may set slack fields"
+// +kubebuilder:validation:XValidation:rule="!has(self.victorOps) || (self.type == 'VICTOR_OPS' && has(self.victorOps))",message="only VICTOR_OPS type may set victorOps fields"
+// +kubebuilder:validation:XValidation:rule="!has(self.webhook) || (self.type == 'WEBHOOK' && has(self.webhook))",message="only WEBHOOK type may set webhook fields"
+type AtlasIntegrationSpec struct {
+	ProjectDualReference `json:",inline"`
+
+	// ID of the integration in Atlas. May be omitted to create a new one.
+	// +kubebuilder:validation:Optional
+	ID *string `json:"id"`
+
+	// Type of the integration
+	// +kubebuilder:validation:Enum:=DATADOG;MICROSOFT_TEAMS;NEW_RELIC;OPS_GENIE;PAGER_DUTY;PROMETHEUS;SLACK;VICTOR_OPS;WEBHOOK
+	// +kubebuilder:validation:Required
+	Type string `json:"type"`
+
+	Datadog *DatadogIntegration `json:"datadog,omitempty"`
+
+	MicrosoftTeams *MicrosoftTeamsIntegration `json:"microsoftTeams,omitempty"`
+
+	NewRelic *NewRelicIntegration `json:"newRelic,omitempty"`
+
+	OpsGenie *OpsGenieIntegration `json:"opsGenie,omitempty"`
+
+	PagerDuty *PagerDutyIntegration `json:"pagerDuty,omitempty"`
+
+	Prometheus *PrometheusIntegration `json:"prometheus,omitempty"`
+
+	Slack *SlackIntegration `json:"slack,omitempty"`
+
+	VictorOps *VictorOpsIntegration `json:"victorOps,omitempty"`
+
+	Webhook *WebhookIntegration `json:"webhook,omitempty"`
+}
+
+type DatadogIntegration struct {
+	// APIKeySecret is the name of a secret containing the datadog api key
+	// +kubebuilder:validation:Required
+	APIKeySecret string `json:"apiKeySecret"`
+
+	// Region is the Datadog region
+	// +kubebuilder:validation:Required
+	Region string `json:"region"`
+
+	// SendCollectionLatencyMetrics flags whether or not to send collection latency metrics
+	// +kubebuilder:validation:Required
+	SendCollectionLatencyMetrics bool `json:"sendCollectionLatencyMetrics"`
```

Comment:
This field is optional and it's recommended to avoid boolean fields (as per DRC-2)


```suggestion
	// +optional
	// +kubebuilder:validation:Enum=enabled;disabled
	// +kubebuilder:default:=disabled
	// Toggle sending collection latency metrics that includes database names and collection names and latency
	// metrics on reads, writes, commands, and transactions.
	CollectionLatencyMetrics *string `json:"collectionLatencyMetrics,omitempty"`
```

Reply:
👌

Reply:
Done on next update